controllers/datafoundationclient.go

```diff
@@ -125,6 +136,27 @@ func (r *dataFoundationClientReconciler) reconcilePhases() (ctrl.Result, error)
 	if !r.offering.DeletionTimestamp.IsZero() {
 		return ctrl.Result{}, nil
 	} else {
+
+		// Checking for CRDs that are installed after the agent was started and add watchers for the new CRDs
+		if !r.ManagedFusionOfferingReconciler.AvailableCRDs[StorageClientCRDName] {
+			crd := apiextensionsv1.CustomResourceDefinition{}
+			crd.Name = StorageClientCRDName
+			if err := r.get(&crd); err != nil {
+				return ctrl.Result{}, err
+			} else {
+				utils.HandleMissingWatchers(StorageClientCRDName)
+			}
+		}
+		if !r.ManagedFusionOfferingReconciler.AvailableCRDs[StorageClassClaimCRDName] {
+			crd := apiextensionsv1.CustomResourceDefinition{}
+			crd.Name = StorageClassClaimCRDName
+			if err := r.get(&crd); err != nil {
+				return ctrl.Result{}, err
+			} else {
+				utils.HandleMissingWatchers(StorageClassClaimCRDName)
+			}
+		}
```

Comment:
@rohan47
Requesting again for the 3rd time (3rd review) - Move this code to the start of reconcilePhases, This code is a prereq to running any of the offering reconcile logic!